centreon-certified/influxdb/influxdb-metrics-apiv2.lua

```diff
@@ -0,0 +1,448 @@
+  self:add()
+  self.sc_logger:debug("[EventQueue:format_metric_service]: end format_metric host")
+end
+
+--------------------------------------------------------------------------------
+---- EventQueue:format_metric_service method
+-- @param metric {table} a single metric data
+--------------------------------------------------------------------------------
+function EventQueue:format_metric_service(metric)
+  self.sc_logger:debug("[EventQueue:format_metric_service]: start format_metric service")
+  local event = self.sc_event.event
+  local metric_key = "metric_" .. mime.b64(tostring(event.host_id) .. ':' .. tostring(event.cache.service.service_id) .. ':' .. tostring(metric.metric_name))
+  event.formated_event = {
+    metric_name = metric.metric_name,
+    metric_value = metric.value,
+    metric_key = metric_key,
+    last_check = event.last_check,
+    host_id = event.host_id,
+    service_id = event.service_id,
+    status = "status value=" .. tostring(event.state) .. ",host_id=" .. tostring(event.host_id) .. ",service_id=" .. tostring(event.cache.service.service_id) .. " " .. tostring(event.last_check)
+  }
+  self:add()
+  self.sc_logger:debug("[EventQueue:format_metric_service]: end format_metric service")
+end
+
+--------------------------------------------------------------------------------
+-- EventQueue:add, add an event to the sending queue
+--------------------------------------------------------------------------------
+function EventQueue:add()
+  -- store event in self.events lists
+  local category = self.sc_event.event.category
+  local element = self.sc_event.event.element
+
+  self.sc_logger:debug("[EventQueue:add]: add event in queue category: " .. tostring(self.sc_params.params.reverse_category_mapping[category])
+    .. " element: " .. tostring(self.sc_params.params.reverse_element_mapping[category][element]))
+
+  self.sc_logger:debug("[EventQueue:add]: queue size before adding event: " .. tostring(#self.sc_flush.queues[category][element].events))
+  self.sc_common:dumper(self.sc_event.event.formated_event)
+  self.sc_flush.queues[category][element].events[#self.sc_flush.queues[category][element].events + 1] = self.sc_event.event.formated_event
+
+  self.sc_logger:info("[EventQueue:add]: queue size is now: " .. tostring(#self.sc_flush.queues[category][element].events)
+    .. ", max is: " .. tostring(self.sc_params.params.max_buffer_size))
+end
+
+--------------------------------------------------------------------------------
+-- EventQueue:build_payload, concatenate data so it is ready to be sent
+-- @param payload {string} json encoded string
+-- @param event {table} the event that is going to be added to the payload
+-- @return payload {string} json encoded string
+--------------------------------------------------------------------------------
+function EventQueue:build_payload(payload, event)
+  if not payload then
+    payload = { event }
+  else
+    table.insert(payload, event)
+  end
+  return payload
+end
```

Comment:
```suggestion
function EventQueue:build_payload(payload, event)
  if not payload then
    payload = event
  else
    payload = payload .. "\n" .. event
  end

  return payload
end
```

this is the same data structure in influxdb1 and influxdb2 so the build_payload function can be written like that